core/dbt/task/run.py

```diff
@@ -93,11 +93,16 @@ def get_execution_status(sql: str, adapter: BaseAdapter) -> Tuple[RunStatus, str
         response, _ = adapter.execute(sql, auto_begin=False, fetch=False)
         status = RunStatus.Success
         message = response._message
+    except (KeyboardInterrupt, SystemExit):
```

Comment:
Does this mean if we cancel during get_execution_status no result file will be written?

Reply:
Correct. We could re-catch the `KeyboardInterrupt / SystemExit` in [after_run](https://github.com/dbt-labs/dbt-core/blob/dd77210756e6432ef2737ab59a65c238e3abfdda/core/dbt/task/run.py#L775) and do something similar to what is done in [execute_nodes](https://github.com/dbt-labs/dbt-core/blob/dd77210756e6432ef2737ab59a65c238e3abfdda/core/dbt/task/runnable.py#L421).